sql/sql_window.cc

```diff
@@ -275,9 +275,12 @@ setup_windows(THD *thd, Ref_ptr_array ref_pointer_array, TABLE_LIST *tables,
       /*
         "The declared type of SK shall be numeric, datetime, or interval"
         we don't support datetime or interval, yet.
+	But it really only matters if the frame is bounded.
       */
       Item_result rtype= win_spec->order_list->first->item[0]->result_type();
-      if (rtype != REAL_RESULT && rtype != INT_RESULT && 
+      if (!win_spec->window_frame->top_bound->is_unbounded() &&
+          !win_spec->window_frame->bottom_bound->is_unbounded() &&
```

Comment:
Do you want the restriction to be applied when only one frame is unbounded or both? If the latter, I guess the new condition must look like 
```
!(win_spec->window_frame->top_bound->is_unbounded() &&
          win_spec->window_frame->bottom_bound->is_unbounded())
```

Reply:
both, so done.